### PR TITLE
Fix tabs state issue v2.1.3

### PR DESCRIPTION
**Pull request by assafchamoy**

### CODE REVIEW

File: src/discrete-layer/components/catalog-tree/catalog-tree.tsx [hunk 1]
```diff
@@ -92,7 +92,7 @@ export const CatalogTreeComponent: React.FC<CatalogTreeComponentProps> = observe
       if (refresh) {
         setIsDataLoading(true);
         void store.catalogTreeStore.initTree();
-        void store.discreteLayersStore.searchParams.resetLocation();
+        // void store.discreteLayersStore.searchParams.resetLocation();
```
Comment (alebinson) [inline src/discrete-layer/components/catalog-tree/catalog-tree.tsx:95]: think worth to remove comment

File: src/discrete-layer/views/discrete-layer-view.tsx [hunk 2]
```diff
@@ -105,7 +105,8 @@ const DRAWING_FINAL_MATERIAL_OPACITY = 0.8;
 const DRAWING_MATERIAL_COLOR = CesiumColor.YELLOW.withAlpha(DRAWING_MATERIAL_OPACITY);
 const DRAWING_FINAL_MATERIAL = new CesiumPolylineDashMaterialProperty({
   color: CesiumColor.DARKSLATEGRAY.withAlpha(DRAWING_FINAL_MATERIAL_OPACITY), //new CesiumColor( 116, 135, 136, 1),
-  dashLength: 5
+  // color: CesiumColor.MAGENTA.withAlpha(1), //new CesiumColor( 116, 135, 136, 1),
```
Comment (alebinson) [inline src/discrete-layer/views/discrete-layer-view.tsx:108]: think worth to remove comment

File: src/discrete-layer/models/discreteLayersStore.ts [hunk 3]
```diff
@@ -221,12 +227,18 @@ export const discreteLayersStore = ModelBase
...
 
-    function setTabviewData(tabView: TabViews): void {
+    function setTabviewData(tabView: TabViews, customLayersImages?: ILayerImage[]): void {
       if (self.tabViews) {
         const idxTabViewToUpdate = self.tabViews.findIndex((tab) => tab.idx === tabView);
 
-        self.tabViews[idxTabViewToUpdate].selectedLayer = self.selectedLayer ? { ...self.selectedLayer } : undefined;
-        self.tabViews[idxTabViewToUpdate].layersImages = self.layersImages ? [ ...self.layersImages ]: [];
+        if(customLayersImages) {
+          const preparedLayersImages = getPreparedLayersImages([...customLayersImages]);
```
Comment (alebinson) [inline src/discrete-layer/models/discreteLayersStore.ts:235]: function setTabviewData() called number of times with already "prepared" data think it's not "biggy", is it ok from your side?

Comment (assafchamoy) [reply src/discrete-layer/models/discreteLayersStore.ts:235]: This is just the case where we want to apply custom layers images directly to the tab view (When we change recordType), so just in that case we need to prepare the data before the insertion. Even if we send in already prepared data it wont cause problems besides calling the method again redundantly.

**Review by alebinson — APPROVED** [review]
gr8!

**Review by alebinson — APPROVED** [review]
gr8!

**Pull request merged**